### PR TITLE
commands/.../up/local: fix kubeconfig and namespace handling

### LINKED ISSUE #980: Repect Kubeconfig variable
## Feature Request

**Is your feature request related to a problem? Please describe.**

KUBECONFIG environment variable doesn't  seem to be respected, it only use .kube/config unless --kubeconfig flag is passed,

**Describe the solution you'd like**

- Use .kube/config as default
- Respect KUBECONFIG variable if there.
- let the user override it with --kubeconfig,

### PR DESCRIPTION
**Pull request by AlexNPavel**
**Description of the change:** This PR allows locally run operators ran via `up local` to use the correct default kubeconfig as well as use the default namespace specified by that kubeconfig.

Since all operator types now use the controller-runtime's config getter, we can set the command's environment to match `os.Environ()` and only set the kubeconfig env var if a user explicitly specifies the path via the `--kubeconfig` flag. This allows the operator to select the appropriate kubeconfig. We also now get the default namespace set in the kubeconfig using `GetKubeconfigAndNamespace()` instead of defaulting to `default`

**Motivation for the change:** Closes #980

Since this is a bugfix, I will cherrypick this fix to backport to the 0.4.x branch once merged.

### CODE REVIEW

File: commands/operator-sdk/cmd/up/local.go [hunk 1]
```diff
@@ -52,7 +52,9 @@ kubernetes cluster using a kubeconfig file.
 
 	upLocalCmd.Flags().StringVar(&kubeConfig, "kubeconfig", "", "The file path to kubernetes configuration file; defaults to $HOME/.kube/config")
 	upLocalCmd.Flags().StringVar(&operatorFlags, "operator-flags", "", "The flags that the operator needs. Example: \"--flag1 value1 --flag2=value2\"")
-	upLocalCmd.Flags().StringVar(&namespace, "namespace", "default", "The namespace where the operator watches for changes.")
+	// a user can set this to "" to watch all namespaces, so we need a different, invalid namespace that we can detect when
+	// deciding whether we need to get the kubeconfig's default namespace or not
+	upLocalCmd.Flags().StringVar(&namespace, "namespace", "#invalid#", "The namespace where the operator watches for changes.")
```
Comment (LiliC) [inline commands/operator-sdk/cmd/up/local.go]: My thinking on this: Would it make more sense to do `#all#` when the user wants to watch for all namespaces? `""` makes me think of blank for me not all ns.

File: commands/operator-sdk/cmd/up/local.go [hunk 2]
```diff
@@ -77,10 +79,18 @@ const (
 )
 
 func upLocalFunc(cmd *cobra.Command, args []string) {
-	mustKubeConfig()
-
 	log.Info("Running the operator locally.")
 
+	// get default namespace to watch if unset
+	if namespace == "#invalid#" {
+		_, defaultNamespace, err := k8sInternal.GetKubeconfigAndNamespace(kubeConfig)
```
Comment (LiliC) [inline commands/operator-sdk/cmd/up/local.go]: What if namespace is set should we still setup the getting of kubeconfig part? I could be missing something as well!

File: commands/operator-sdk/cmd/up/local.go [hunk 3]
```diff
@@ -52,7 +52,9 @@ kubernetes cluster using a kubeconfig file.
 
 	upLocalCmd.Flags().StringVar(&kubeConfig, "kubeconfig", "", "The file path to kubernetes configuration file; defaults to $HOME/.kube/config")
```
Comment (LiliC) [inline commands/operator-sdk/cmd/up/local.go]: Can we add a comment that `$KUBECONFIG` will be respected if it set.

File: commands/operator-sdk/cmd/up/local.go [hunk 4]
```diff
@@ -77,10 +79,18 @@ const (
 )
```
Comment (LiliC) [inline commands/operator-sdk/cmd/up/local.go]: On line 76: Is `defaultConfigPath` used anywhere?

File: commands/operator-sdk/cmd/up/local.go (hunk 1 above)
Comment (AlexNPavel) [reply commands/operator-sdk/cmd/up/local.go]: Kubernetes/client-go defines `""` as all namespaces, so we're just following that convention

File: commands/operator-sdk/cmd/up/local.go (hunk 2 above)
Comment (AlexNPavel) [reply commands/operator-sdk/cmd/up/local.go]: No, here we only use that function just to get the default namespace specified by the kubeconfig and we discard the kubeconfig itself using `_`. The operators get the kubeconfigs themselves, so we don't get the kubeconfig.

File: commands/operator-sdk/cmd/up/local.go (hunk 4 above)
Comment (AlexNPavel) [reply commands/operator-sdk/cmd/up/local.go]: Nope, removed

File: commands/operator-sdk/cmd/up/local.go (hunk 1 above)
Comment (estroz) [reply commands/operator-sdk/cmd/up/local.go]: I believe you can use the [`cmd.Flag("namespace").Changed`](https://github.com/spf13/cobra/issues/453#issuecomment-303548563) boolean below to detect whether the user has supplied a namespace.

Comment (AlexNPavel) [reply commands/operator-sdk/cmd/up/local.go]: @estroz Nice, didn't know about that. I'll use that instead to make this a bit cleaner

File: commands/operator-sdk/cmd/up/local.go (hunk 2 above)
Comment (LiliC) [reply commands/operator-sdk/cmd/up/local.go]: Ah that makes sense, thanks for clarifying!

**Pull request merged**